### PR TITLE
Adding In-line Edit Design Documentation

### PR DESCRIPTION
**Pull request by beanh66**
The In-line Edit pattern can be used in conjunction with a List View or a Table view to allow the user to create or edit an item without navigating to another view. 

For more information on the in-line edit pattern, previous PatternFly discussions can be found here: https://www.redhat.com/archives/patternfly/2016-October/msg00061.html

### CODE REVIEW

File: pattern-library/Inline_Edit/Inline_edit-overview.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# In-Line Edit
+
+## Overview
+
+In-line Edit can be used in conjunction with a List View or a Table view to allow the user to create or edit an item without navigating to another view.  Use this when all editable elements can be viewed within the row or expanded row. In-line Edit saves the user the need to open a modal window or navigate to a new view for simple object creation or edit actions.
```
Comment (LHinson) [inline pattern-library/Inline_Edit/Inline_edit-overview.md]: "Table View" vs "Table view"

Comment (LHinson) [inline pattern-library/Inline_Edit/Inline_edit-overview.md]: For your consideration... It says "when all editable elements can be viewed within the row or expanded row". Would another option be when there is a "quick add" version? Word Press has something like this for posts. It gives you some quick options up front. Ultimately you would have to navigate to another view for the full set of options.

**Comment by LHinson** [comment]
Overall very nice job @beanh66. I see this was added directly under pattern library. We will need to decide where this best fit on the site so that we can add it to the appropriate place within the github directories. 

It could be considered a "control" possibly. Thoughts @beanh66 @andresgalante @serenamarie125

**Comment by thomasmaas** [comment]
As inline editing is all about being able to make quick edits, maybe we should say something about focussing the first form field and keyboard navigation?

File: pattern-library/Inline_Edit/Inline_edit-overview.md (hunk 1 above)
Comment (beanh66) [inline pattern-library/Inline_Edit/Inline_edit-overview.md]: @LHinson that's a good point and I think worth noting that as a possible use case as well! What do you think @mcarrano @vconzola ?

**Comment by beanh66** [comment]
@LHinson Yes, sorry about that. I didn't realize I could dictate where the pattern goes based on my submission. I definitely did not create the right folder structure, but let me know the best way to reconcile! I agree, I would think this belongs in "Forms and Controls" and I figured we can add links to this pattern in the List View and Table View sections. @mcarrano @vconzola, do you agree?

**Comment by beanh66** [comment]
@thomasmaas Thanks for that note, I agree for ease of use and will add it to the documentation!

**Comment by mcarrano** [comment]
As for where this should go in the folder structure, I was thinking that it belonged in "Content Views."  I say that because this is a specific extension of list and table views, only.  For me, that's the first place I would look for it.  But linking from the List and Table View patterns is also acceptable.

File: pattern-library/Inline_Edit/Inline_edit-design.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+2. **Multi-Select:** Only one row can be edited at a time. If multi-select behavior is supported, and multiple rows are selected the row selections should persist when a single row is in edit mode but the associated actions should be disabled and the blue background highlighting of the selected rows should disappear. Upon exit of edit mode the row selections remain and the global actions are enabled. Upon exit of edit mode the row selections remain and the global actions are enabled.
+3. **Edit:** action may be accessed through a button, an icon, or the kebab menu and puts the row into an editable state.
+
+
+## In-line Edit in a Table View
+![Image of standard list view showing Edit action.](Table_01.png)
+
+1. **Row Selection:** Only one row can be edited at a time. Edit puts an entire row into edit mode. The row in edit mode should be visually distinct from surrounding rows.
+2. **Save and Cancel (bottom):** buttons are displayed in a tab under the bottom right corner of the row. The check mark should be used for save and the X for cancel. Save is disabled until a change is made.
```
Comment (LHinson) [inline pattern-library/Inline_Edit/Inline_edit-design.md]: Consolidate the "Save and Cancel" into one item and bullet point the one for bottom and then the one for top

File: pattern-library/Inline_Edit/Inline_edit-overview.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+* Only one row in a list or a table can be edited at a time.  Users commit to or cancel changes by clicking **Save** or **Cancel**.
+
+* Items can be added to the top of a list or a table by including an **Add** or **Create** action.  
+
+## In-Line Edit for List view
+
+![Image of List View](List_Edit_00.png)
+
+## In-Line Edit for Table view
```
Comment (LHinson) [inline pattern-library/Inline_Edit/Inline_edit-overview.md]: Table View vs Table view

File: pattern-library/Inline_Edit/Inline_edit-design.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+1. **Create:** adds a row at the top of the list, in edit mode. All fields in the new row are editable. Default values for all fields are recommended.
+2. **Save and Cancel:** buttons are available on the right end of the row. If user selects Save (check) without entering values for all fields an error is displayed. If user selects Cancel the new row is deleted.
+
+## In-line Edit
+![Image of list with edit action.](List_Edit_02.png)
+
+1. **Row Selection:** Only one row can be edited at a time. Edit puts an entire row into edit mode. The row in edit mode should be visually distinct from surrounding rows.
+2. **Editable Fields:**  in the selected row should change from read only into an editable state such as a text field or combobox.  
+3. **Non-editable Fields:** in the selected row should not change in appearance.
```
Comment (LHinson) [inline pattern-library/Inline_Edit/Inline_edit-design.md]: "Non-Editable" vs "Non-editable"

File: pattern-library/forms-and-controls/inline-edit/Inline_edit-design.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+1. **Create:** adds a row at the top of the list, in edit mode. All fields in the new row are editable. Default values for all fields are recommended.
+2. **Save and Cancel:** buttons are available on the right end of the row. If user selects Save (check) without entering values for all fields an error is displayed. If user selects Cancel the new row is deleted.
+
+## In-line Edit
+![Image of list with edit action.](img/List_Edit_02.png)
+
+1. **Row Selection:** Only one row can be edited at a time. Edit puts an entire row into edit mode. The row in edit mode should be visually distinct from surrounding rows. The first editable field should be in focus with an option to tab through all remaining editable fields with keyboard navigation.
+2. **Editable Fields:**  in the selected row should change from read only into an editable state such as a text field or combobox.  
+3. **Non-editable Fields:** in the selected row should not change in appearance.
```
Comment (LHinson) [inline pattern-library/forms-and-controls/inline-edit/Inline_edit-design.md]: "Non-Editable" vs "Non-editable"

File: pattern-library/forms-and-controls/inline-edit/Inline_edit-design.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+2. **Multi-Select:** Only one row can be edited at a time. If multi-select behavior is supported, and multiple rows are selected the row selections should persist when a single row is in edit mode but the associated actions should be disabled and the blue background highlighting of the selected rows should disappear. Upon exit of edit mode the row selections remain and the global actions are enabled. Upon exit of edit mode the row selections remain and the global actions are enabled.
+3. **Edit:** action may be accessed through a button, an icon, or the kebab menu and puts the row into an editable state.
+
+
+## In-line Edit in a Table View
+![Image of standard list view showing Edit action.](img/Table_01.png)
+
+1. **Row Selection:** Only one row can be edited at a time. Edit puts an entire row into edit mode. The row in edit mode should be visually distinct from surrounding rows.
+2. **Save and Cancel (bottom):** buttons are displayed in a tab under the bottom right corner of the row. The check mark should be used for save and the X for cancel. Save is disabled until a change is made.
```
Comment (LHinson) [inline pattern-library/forms-and-controls/inline-edit/Inline_edit-design.md]: Consolidate the save and cancel items into one label and a bulleted list

File: pattern-library/Inline_Edit/Inline_edit-design.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
+# In-line Edit
+
+## In-line Create and Edit
+![Image of list with newly created row.](List_Edit_01.png)
+
+1. **Create:** adds a row at the top of the list, in edit mode. All fields in the new row are editable. Default values for all fields are recommended.
```
Comment (LHinson) [inline pattern-library/Inline_Edit/Inline_edit-design.md]: use completed sentences so it should read "Create add a row..."

File: pattern-library/Inline_Edit/Inline_edit-design.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+2. **Save and Cancel:** buttons are available on the right end of the row. If user selects Save (check) without entering values for all fields an error is displayed. If user selects Cancel the new row is deleted.
```
Comment (LHinson) [inline pattern-library/Inline_Edit/Inline_edit-design.md]: "The save and cancel buttons..."

File: pattern-library/Inline_Edit/Inline_edit-design.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+## In-line Edit
+![Image of list with edit action.](List_Edit_02.png)
+
+1. **Row Selection:** Only one row can be edited at a time. Edit puts an entire row into edit mode. The row in edit mode should be visually distinct from surrounding rows.
+2. **Editable Fields:**  in the selected row should change from read only into an editable state such as a text field or combobox.
```
Comment (LHinson) [inline pattern-library/Inline_Edit/Inline_edit-design.md]: The editable field in the selected row...

File: pattern-library/Inline_Edit/Inline_edit-design.md [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+1. **Create:** adds a row at the top of the list, in edit mode. All fields in the new row are editable. Default values for all fields are recommended.
+2. **Save and Cancel:** buttons are available on the right end of the row. If user selects Save (check) without entering values for all fields an error is displayed. If user selects Cancel the new row is deleted.
+
+## In-line Edit
+![Image of list with edit action.](List_Edit_02.png)
+
+1. **Row Selection:** Only one row can be edited at a time. Edit puts an entire row into edit mode. The row in edit mode should be visually distinct from surrounding rows.
+2. **Editable Fields:**  in the selected row should change from read only into an editable state such as a text field or combobox.  
+3. **Non-editable Fields:** in the selected row should not change in appearance.
+4. **Save and Cancel:** buttons are available on the right end of the row. The check mark (Save) and X (Cancel) should replace the kebab in the editable row. Save is disabled until a change is made.
```
Comment (LHinson) [inline pattern-library/Inline_Edit/Inline_edit-design.md]: The save and cancel buttons...

File: pattern-library/Inline_Edit/Inline_edit-design.md [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+5. **Edit:** puts the row into an editable state. This action may be accessed through a button, an icon, or the kebab menu.
```
Comment (LHinson) [inline pattern-library/Inline_Edit/Inline_edit-design.md]: Edit puts...

File: pattern-library/Inline_Edit/Inline_edit-design.md [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+6. **Click Away:** Users can click away from a row in edit mode. When a new row has been created, If changes have not been made, then clicking away acts like cancel - edit mode is exited without confirmation. If changes have been made, then a dialog asking user if they want to save changes is displayed before exiting edit mode.
```
Comment (LHinson) [inline pattern-library/Inline_Edit/Inline_edit-design.md]: "Users can click away from a row in edit mode. When a new row has been created, If changes have not been made, then clicking away acts like cancel - edit mode is exited without confirmation."  <<< this sentence needs some cleanup. It's a bit confusing

File: pattern-library/Inline_Edit/Inline_edit-design.md [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+
+
+## In-line Edit in an Expanded List View
+![Image of list with expanded row and edit action.](List_Edit_03.png)
+
+1. **Row Expansion:** If expansion behavior is supported, when a row is in edit mode, it should automatically expand. Editable fields that are not visible in the standard list can be edited in the expanded region. The X (close control) for the expanded region should be hidden; the expanded region cannot be collapsed in edit mode. The row should be returned back to the collapsed state once the user saves or cancels out of edit mode.
+2. **Multi-Select:** Only one row can be edited at a time. If multi-select behavior is supported, and multiple rows are selected the row selections should persist when a single row is in edit mode but the associated actions should be disabled and the blue background highlighting of the selected rows should disappear. Upon exit of edit mode the row selections remain and the global actions are enabled. Upon exit of edit mode the row selections remain and the global actions are enabled.
+3. **Edit:** action may be accessed through a button, an icon, or the kebab menu and puts the row into an editable state.
```
Comment (LHinson) [inline pattern-library/Inline_Edit/Inline_edit-design.md]: need a complete sentence

**Comment by LHinson** [comment]
@beanh66 I've made some editing comments. Let me know if you have any questions!

**Comment by beanh66** [comment]
@LHinson Thanks for the comments, I pushed a bunch of small changes yesterday if you want to take a look when you get a chance!

**Comment by LHinson** [comment]
bitmoji

**Pull request merged**